Adjust: Asset and Enemy Card Interaction in Play Area

SHOrTEVA opened the pull request:
Modify the interaction for asset and enemy cards in the play area to allow a single click for triggering the only one option, eliminating the need for the button to pop out.

halogenandtoast committed:
Add telemetry plugin

halogenandtoast commented:
I am not sure I'd want to do this. Sometimes it might be non-obvious what the reaction is (for instance Hatchet (1), in which case the tooltip helps).

halogenandtoast commented:
> If it is only available on mobile devices, can you accept that? 

Maybe? I wish there was some other way to capture this kind of interaction without getting rid of the buttons. For now if you make it mobile only I'll merge it in until I can come up with a better option.

SHOrTEVA commented:
> I'll
Since it might be unclear what the reaction is, I could make it work like a ‘Force’ button: if only one reaction is triggered, a button pops up. Pressing either the button or the card activates the effect, while long-pressing the button shows the tooltip. Do you think this is a good approach?

halogenandtoast commented:
If you mean one reaction across all cards than this is fine. It would need to be since AbilitiesMenu operates as a singleton.

halogenandtoast committed:
Return to Union and Disillusion

halogenandtoast committed:
Fixes

halogenandtoast committed:
Use b sides

halogenandtoast committed:
Modernize In The Clutches of Chaos

halogenandtoast committed:
Return to In the Clutches of Chaos progress

halogenandtoast committed:
Return to Southside

halogenandtoast committed:
Digest Frech Film Fatale

halogenandtoast committed:
Return to Merchant District

halogenandtoast committed:
Fixes

halogenandtoast committed:
Return to In The Clutches of Chaos

halogenandtoast committed:
Do not error on event already being gone

halogenandtoast committed:
Fix enemy defeat

halogenandtoast committed:
Handle PerMove windows better

halogenandtoast committed:
Fix Atlach Nacha

halogenandtoast committed:
Modernize Before the Black Throne Cards

halogenandtoast committed:
Finish updating Before the Black Throne

halogenandtoast committed:
Campaign fixes

halogenandtoast committed:
Return to Before the Black Throne and Epilogue

halogenandtoast committed:
Update beta flag

halogenandtoast committed:
Fix xp copy

halogenandtoast committed:
Cleanup, and fix tests

halogenandtoast committed:
Fixes

Commit:
Update Trespasser.hs (#2801)

halogenandtoast committed:
Fix Overgrown Barn

Add check enemy engagement since it might not have happened.

halogenandtoast committed:
The Scarlet Keys Init

halogenandtoast committed:
Riddles and Rain Intro

halogenandtoast committed:
WIP

halogenandtoast committed:
Make healable consider assigned damage, Fixes Bandages

halogenandtoast committed:
Fixes

halogenandtoast committed:
Updates

halogenandtoast committed:
Use PerTest window

halogenandtoast committed:
Fixes

halogenandtoast committed:
Calendar Work

halogenandtoast committed:
Change compiler name

halogenandtoast committed:
Remove scss

halogenandtoast committed:
Add back in scss using sass-embedded

halogenandtoast committed:
CSS Fixes

halogenandtoast committed:
More css fixes

halogenandtoast committed:
Fix test suite

halogenandtoast committed:
Fix skill test styles

YouriAndropov committed:
New translations added (#2821)

halogenandtoast committed:
Fixes

halogenandtoast committed:
Fix setup modifiers

halogenandtoast committed:
Underworld Market adjustments

halogenandtoast committed:
TSK work

YouriAndropov committed:
Fix 2 cards in Dream-Eaters (#2840)

halogenandtoast committed:
Concealed progress

halogenandtoast committed:
Fixes

halogenandtoast committed:
Fix A Cosmic Journey

halogenandtoast committed:
Endless Descent fixes

halogenandtoast committed:
Fix In The Shadows

halogenandtoast committed:
Fixes

halogenandtoast committed:
Fix PerSpawn

halogenandtoast committed:
Update fr digest

halogenandtoast committed:
Fixes

halogenandtoast committed:
More concealed work

halogenandtoast committed:
More concealed work

halogenandtoast committed:
Concealed attack work

halogenandtoast committed:
Handle automatic evasion for concealed

halogenandtoast committed:
More concealed work

halogenandtoast committed:
More Riddles and Rain

halogenandtoast committed:
Some encounter sets

halogenandtoast committed:
Alots of TSK

halogenandtoast committed:
Hover over labels

halogenandtoast committed:
fix modal

halogenandtoast committed:
Some fixes

halogenandtoast committed:
Fixes

halogenandtoast committed:
Allow empty space for connections

halogenandtoast committed:
Fix Bum Rush, Fix City of the Elder things setup

halogenandtoast committed:
Some EOTE fixes

halogenandtoast committed:
Update the seal weakens behavior

halogenandtoast committed:
Fixes

halogenandtoast committed:
Timing

YouriAndropov committed:
Fix card inversion (#2886)

halogenandtoast committed:
Fixes

halogenandtoast committed:
Some TFA fixes

halogenandtoast committed:
Mark modifiers as active during setup (TFA)

halogenandtoast committed:
Fix key

halogenandtoast committed:
Fixes

halogenandtoast committed:
Fixes

halogenandtoast committed:
Fixes

halogenandtoast committed:
Oops

halogenandtoast committed:
Fix connections

halogenandtoast committed:
More fixes

halogenandtoast committed:
Map updates

halogenandtoast committed:
Fix Waterfall

halogenandtoast committed:
Make FailSkillTest commit cards

halogenandtoast committed:
Fix window to match against skill test

halogenandtoast committed:
Update PassSkillTest to commit cards

halogenandtoast committed:
Destiny reading needs to happen before investigator setup

The Hanged Man needs to affect the mulligan so it must be before.

halogenandtoast committed:
Unhallowed Land should only target Ally cards

halogenandtoast committed:
Make horror/damage cost the last cost to be applied

This prevents costs becoming unpayable due to the investigator being
defeated mid-payment.

halogenandtoast committed:
Hidden Agendas: Wine Cellar should flip to spectral

halogenandtoast committed:
Fix Brown Jenkin

Should not include hidden cards when determining how many cards to draw

halogenandtoast committed:
Move When Game Ends window to before Scenario Resolution

Ensures that XP modifiers from end of game effects take effect

halogenandtoast committed:
Cleanup

halogenandtoast committed:
Fix out of play skill modifiers

Seal of the Elder Sign (5) would affect the rest of the game when
played.

halogenandtoast committed:
Fix specs

halogenandtoast committed:
Progress on travel

halogenandtoast committed:
More map work

halogenandtoast committed:
Dossier

halogenandtoast committed:
Small adjustments to map

derornos committed:
feat: update ko translations (#2902)

halogenandtoast committed:
Dead Heat Progress

halogenandtoast committed:
Log changes

halogenandtoast committed:
Dead Heat work

halogenandtoast committed:
Some more Dead Heat / TSK

halogenandtoast committed:
Some Key work

halogenandtoast committed:
Some more TSK work

halogenandtoast committed:
Wrap up Dead Heat

halogenandtoast committed:
Fix Swarm Enemies

halogenandtoast committed:
Fix Stalking Maticore

halogenandtoast committed:
Formatting

halogenandtoast committed:
Fix Occult Lexicon

halogenandtoast committed:
Fix addCardEntity effects

Because addCardEntity reused the card id we'd have situations , like
Robert Castaigne (4) and British Bulldog (2), where the card being added
already created a card entity due to having an out of play effect and so
it would have two "shadow" copies. This resulted in abilities being
activated twice (once for the shadow copy created via addCardEntity and
once for the out of play effects copy).

We fix that by giving addCardEntity entities a randomly generated id
rather than reusing the card id.

halogenandtoast committed:
Fix Richard Upton Pickman

halogenandtoast committed:
Make Daredevil interaction more noticeable

halogenandtoast committed:
Add ability to highlight ability targets

halogenandtoast committed:
Nightgaunt steed prompt

halogenandtoast committed:
Some cosmos changes

halogenandtoast committed:
Some adjustments to pathing

halogenandtoast committed:
Fix Disappearance at the Twilight Estate

halogenandtoast committed:
Update Terror Unleashed

halogenandtoast committed:
Fix Cloud Forest

halogenandtoast committed:
Fix clue highlight

halogenandtoast committed:
Fix Need for Knowledge

halogenandtoast committed:
Format Holy Spear (5)

halogenandtoast committed:
Fix Unstable Vortex's Shroud and Clue Count

halogenandtoast committed:
Fix Riverside Temple's Chalk Check

Needs to check all investigators, not just the triggering investigator

halogenandtoast committed:
Fix skill choice formatting

halogenandtoast committed:
Fix Amanda Sharpe Forced Interaction

Additional costs are not paid when forced.

halogenandtoast committed:
Don't show action specific modifiers when action mismatches

halogenandtoast committed:
Fix the heart of madness elderThing i18n key

halogenandtoast committed:
Possessed Treachery should be a weakness

halogenandtoast committed:
Partner choice needs to recurse

Selecting Takada would exit early

halogenandtoast committed:
Fix Green Soapstone

Adds a new skill test window for general "attacking" checks

halogenandtoast committed:
Fix Collapse The Pylons

Needs to remove the Agenda

halogenandtoast committed:
Fix The Secret Must Be Kept

Difficulty needs to consider return to

halogenandtoast committed:
Fix Tempo Mayor (175)

Need to be at location to use ability

halogenandtoast committed:
Fix Delores Gadling

Can only ignore AOO of Humanoid enemies

halogenandtoast committed:
Fix Lucky Penny

Must be controlled for ability to trigger

halogenandtoast committed:
Trigger after enter windows with fishing vessel

halogenandtoast committed:
Better flood handling when in Fishing Vessel

halogenandtoast committed:
Add SanguineShadows choice

halogenandtoast committed:
Allow act advance during any fast player window

halogenandtoast committed:
Sanguine Shadows work

halogenandtoast committed:
Sanguine Shadows, Act and Agenda work

halogenandtoast committed:
Sanguine Shadows token effects

halogenandtoast committed:
Where Is She

halogenandtoast committed:
Extract some more helpers

halogenandtoast committed:
Finish locations

halogenandtoast committed:
Some enemies

halogenandtoast committed:
La Chica Roja

halogenandtoast committed:
More Sanguine Shadows work

halogenandtoast committed:
Sanguine Shadows fixes

halogenandtoast committed:
Embark

halogenandtoast committed:
Interlude 20

halogenandtoast committed:
Interlude 26

halogenandtoast committed:
Silver Moth needs to go to threat area

halogenandtoast committed:
Discover the truth should measure nearest from clinton

halogenandtoast committed:
Missing unrevealed abilities on return to historical societies

halogenandtoast committed:
Fix Roland Banks / Stubborn Detective interaction

halogenandtoast committed:
Some Dealings in the Dark Porgress

YouriAndropov committed:
TFA: FR locale completed. (#2952)

halogenandtoast committed:
Dealing in Dark WIP

halogenandtoast committed:
Dealing in the Dark progress

halogenandtoast committed:
Rooms tracker

halogenandtoast committed:
Fix French translation

halogenandtoast committed:
Show more game details

halogenandtoast committed:
Show room data on main admin dashboard

halogenandtoast committed:
Why is this here

halogenandtoast committed:
Debug rooms

halogenandtoast committed:
Room updates

halogenandtoast committed:
Manual room deletion

halogenandtoast committed:
If page loads missing page, delete room

halogenandtoast committed:
Hmm

halogenandtoast committed:
Manual room deletion

halogenandtoast committed:
More room tracking adjustments

halogenandtoast committed:
Async fix

halogenandtoast committed:
Some work on different locking mechanisms

halogenandtoast committed:
Move plugin

halogenandtoast committed:
Plugin all the places

halogenandtoast committed:
Update again

halogenandtoast committed:
Tracing

halogenandtoast committed:
Improve playable check performance

halogenandtoast committed:
Fix campaign key

halogenandtoast committed:
More optimizations

halogenandtoast committed:
Remove display flex from scenario-body

Littleku1 committed:
Update heartOfTheElders.json with new transtation (#2989)

Translation to zh

halogenandtoast committed:
Fix ZH translations

halogenandtoast committed:
Conduit is likely not doing anything here

halogenandtoast committed:
Merge branch 'main' into the-scarlet-keys

halogenandtoast committed:
Mark TSK alpha

halogenandtoast committed:
Fix Bearer Only Prey

halogenandtoast committed:
Track location replacement

halogenandtoast committed:
Fix return to continuation

halogenandtoast committed:
Threads of Fate cleanup

halogenandtoast committed:
Display Extra Xp

halogenandtoast committed:
Show unspendable xp in other situations

halogenandtoast committed:
Fix commit logic

halogenandtoast committed:
Fix modifiers popup

halogenandtoast committed:
Fix specs

halogenandtoast committed:
Fix Flux Stabilizer

halogenandtoast committed:
Fix cards with reaction in discard

halogenandtoast committed:
Exchange UI

halogenandtoast committed:
Fix taboo limit on swift reflexes

halogenandtoast committed:
Fix in hand card abilities

halogenandtoast committed:
Track Nahab/Brown Jenkin defeat correctly

halogenandtoast committed:
Delve too deep should be in player order

halogenandtoast committed:
Change hospital debts (and other fast ability) limits

halogenandtoast committed:
Token pool fixes

halogenandtoast committed:
Don't double place clues on act

halogenandtoast committed:
xp text

halogenandtoast committed:
Some fixes and debug options

halogenandtoast committed:
Fix Hall of Rebirth

The active investigator should trigger the ability

halogenandtoast committed:
Fix Hiding Spot

Should not affect Elite enemies

halogenandtoast committed:
Make locationMatches more resilient

halogenandtoast committed:
Fix Lake Xochimilco

Need to check card code for ability comparison

halogenandtoast committed:
Add a bounds to exchanging resources in the UI

Should probably also do the backend

halogenandtoast committed:
Fix possession traitorous from going infinity

halogenandtoast committed:
Add game id to trace

halogenandtoast committed:
Do not double up flipped assets with stories

halogenandtoast committed:
Add missing field

halogenandtoast committed:
Thermos should use supplies

halogenandtoast committed:
Trauma type should be picked if both happen

halogenandtoast committed:
Allow set campaign step via debug

halogenandtoast committed:
Fix Veda Whitsley

halogenandtoast committed:
Fix Safeguard (2)

halogenandtoast committed:
Winged Serpent is double sided

halogenandtoast committed:
Fix many other sided cards

halogenandtoast committed:
Fix Gondola

halogenandtoast committed:
Fix the winged serpent

Pillar instead of resource

halogenandtoast committed:
Fix removing pool specific tokens

YouriAndropov committed:
Midwinter Gala FR + some FR locale fixes (#3047)

halogenandtoast committed:
Fix HOTE R2

halogenandtoast committed:
Fix Return to Heart of the Elders

halogenandtoast committed:
Fix Harbinger

derornos committed:
feat: update ko translations (tfa) (#3061)

halogenandtoast committed:
Fix some ts issues

halogenandtoast committed:
Fix Possessed

halogenandtoast committed:
Fix Bob Jenkins

halogenandtoast committed:
Don't show alpha campaigns when non-alpha user

halogenandtoast committed:
Fix Emerging Deep One

halogenandtoast committed:
Fix Undertow

halogenandtoast committed:
Fix Pay Day (1)

halogenandtoast committed:
Fix Vault of Earthly Demise

halogenandtoast committed:
Fix Father Mateo Parallel on reveal

halogenandtoast committed:
Allow abilities to only avoid AOO from specific enemies

SHOrTEVA committed:
Merge branch 'halogenandtoast:main' into pr-playerArea

halogenandtoast commented:
I think this has diverged too much at this point so I am going to close it. If you want to revisit it let me know. I may continue to try and improve the mobile experience as well.
